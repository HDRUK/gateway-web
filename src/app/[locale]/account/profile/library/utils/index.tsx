import { IconButton } from "@mui/material";
import { createColumnHelper } from "@tanstack/react-table";
import { LibraryListItem, SelectedLibrary } from "@/interfaces/Library";
import Link from "@/components/Link";
import StyledCheckbox from "@/components/StyledCheckbox";
import TooltipIcon from "@/components/TooltipIcon";
import { CheckIcon, DeleteForeverIcon } from "@/consts/icons";
import { RouteName } from "@/consts/routeName";

const columnHelper = createColumnHelper<LibraryListItem>();

const getColumns = ({
    handleSelect,
    handleRemove,
    selected,
    translations,
}: {
    handleSelect: (data: SelectedLibrary) => void;
    handleRemove: (id: string | number) => void;
    selected: SelectedLibrary;
    translations: { [id: string]: string };
}) => [
    columnHelper.display({
        id: "actions",
        meta: { isPinned: true },
        cell: ({ row }) => {
            const {
                id,
                datasetId,
                name,
                dataCustodianId,
                dataCustodian,
                darEnabled,
                darTemplatePublished,
                cohortEnabled,
            } = row.original;
            return (
                <div style={{ textAlign: "center" }}>
                    <StyledCheckbox
                        checked={selected[id]?.selected}
                        onChange={(_e, value) =>
                            handleSelect({
                                [id]: {
                                    selected: value,
                                    datasetId,
                                    name,
                                    teamId: dataCustodianId,
                                    teamName: dataCustodian,
                                    darEnabled,
                                    darTemplatePublished,
                                    cohortEnabled,
                                },
                            })
                        }
                        size="large"
                        sx={{ p: 0 }}
                        iconSx={{ mr: 0 }}
                    />
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),

    columnHelper.display({
        id: "name",
        cell: ({ row: { original } }) => {
            const { datasetId, name } = original;
            const linkHref = `/${RouteName.DATASET_ITEM}/${datasetId}`;

            return (
                <Link href={linkHref}>
                    <div style={{ textAlign: "center" }}>{name}</div>
                </Link>
            );
        },
        header: () => <span>{translations.name}</span>,
        size: 150,
    }),

    columnHelper.display({
        id: "darEnabled",
        cell: ({
            row: {
                original: { darEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {darEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => (
            <TooltipIcon
                buttonSx={{ p: 0 }}
                size="small"
                label={translations.darEnabled}
                content={translations.darEnabled}
            />
        ),
        size: 100,
    }),

    columnHelper.display({
        id: "cohortEnabled",
        cell: ({
            row: {
                original: { cohortEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {cohortEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => <span>{translations.cohortEnabled}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "dataCustodian",
        cell: ({
            row: {
                original: { dataCustodian },
            },
        }) => <div style={{ textAlign: "center" }}>{dataCustodian}</div>,
        header: () => <span>{translations.dataCustodian}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "entityType",
        cell: ({
            row: {
                original: { entityType },
            },
        }) => <div style={{ textAlign: "center" }}>{entityType}</div>,
        header: () => <span>{translations.entityType}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "delete",
        cell: ({ row: { original } }) => {
            return (
                <div style={{ textAlign: "center" }}>
                    <IconButton onClick={() => handleRemove(original.id)}>
                        <DeleteForeverIcon color="primary" />
                    </IconButton>
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),
];

export { getColumns };
