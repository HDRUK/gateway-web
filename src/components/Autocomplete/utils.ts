import { ValueType } from "./Autocomplete";

const getChipLabel = (
    options: { value: string | number; label: string }[],
    value: ValueType
) => {
    if (typeof value === "string") {
        return value;
    }

    return options.find(option => option.value === value)?.label;
};

export { getChipLabel };
