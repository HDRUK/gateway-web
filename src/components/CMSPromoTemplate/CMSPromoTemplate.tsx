"use client";

import { ReactElement } from "react";
import { PageTemplatePromo } from "@/interfaces/Cms";
import Box from "@/components/Box";
import Button from "@/components/Button";
import GradientBanner from "@/components/GradientBanner";
import HTMLContent from "@/components/HTMLContent";
import Link from "@/components/Link";
import "@/styles/wpStyles.css";

const CMSPromoTemplate = ({
    content,
    ctaOverrideComponent,
}: {
    content: PageTemplatePromo;
    ctaOverrideComponent?: ReactElement;
}) => {
    const {
        template: { promofields },
    } = content;
    const CtaContent =
        ctaOverrideComponent ||
        (promofields.ctaLink && (
            <Link passHref href={promofields.ctaLink.url}>
                <Button sx={{ mt: 3 }}>{promofields.ctaLink.title}</Button>
            </Link>
        ));
    return (
        <div className="wpStyles">
            <GradientBanner title={promofields.bannerTitle} />
            <Box
                sx={{
                    bgcolor: "white",
                    pt: 5,
                    pb: 8,
                    px: 6,
                    display: "flex",
                    alignItems: "center",
                    flexDirection: "column",
                }}>
                <Box
                    sx={{
                        p: 0,
                        display: "flex",
                        alignItems: "center",
                    }}>
                    <Box sx={{ flex: 1 }}>
                        <HTMLContent content={promofields.topLeftPanel} />
                    </Box>
                    <Box sx={{ flex: 1 }}>
                        {promofields.topRightPanel && (
                            <HTMLContent content={promofields.topRightPanel} />
                        )}
                    </Box>
                </Box>
                {CtaContent}
            </Box>
            <Box
                sx={{
                    bgcolor: "primary.main",
                    color: "white",
                    display: "flex",
                    flexDirection: "column",
                    alignItems: "center",
                    pt: 5,
                    pb: 8,
                    px: 6,
                }}>
                <Box>
                    <HTMLContent content={promofields.middlePanel} />
                </Box>
                {CtaContent}
            </Box>
            {promofields.bottomPanel && (
                <Box
                    sx={{
                        display: "flex",
                        bgcolor: "white",
                        alignItems: "center",
                        flexDirection: "column",
                        pt: 5,
                        pb: 8,
                        px: 6,
                    }}>
                    <Box sx={{ display: { laptop: "flex" } }}>
                        <Box sx={{ flex: 1 }}>
                            <HTMLContent content={promofields.bottomPanel} />
                        </Box>
                    </Box>
                    {CtaContent}
                </Box>
            )}
        </div>
    );
};

export default CMSPromoTemplate;
