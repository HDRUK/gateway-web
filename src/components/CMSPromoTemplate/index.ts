import CMSPromoTemplate from "./CMSPromoTemplate";

export default CMSPromoTemplate;
