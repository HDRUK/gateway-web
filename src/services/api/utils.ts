import { NotificationOptions } from "@/interfaces/Api";
import { AxiosResponse } from "axios";
import { Error } from "@/interfaces/Error";
import notificationService from "../notification";

interface ErrorNotificationProps {
    props: NotificationOptions;
    errorResponse: AxiosResponse<Error>;
    method: "delete" | "post" | "put" | "get";
}

const errorNotification = ({
    errorResponse,
    method,
    props,
}: ErrorNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const { data, status } = errorResponse || {};

    const fallbackTitle = i18n.exists(`api:common.error.status.${status}`)
        ? t(`api:common.error.status.${status}`)
        : "There has been an error";

    const title = data?.title || fallbackTitle;
    const message =
        data?.message ||
        t(`api:common.error.${method}.message`, {
            item: props.itemName || "Item",
        });

    notificationService.apiError(message, {
        title,
        message,
        errors: data?.errors,
        ...notificationProps,
    });
};

interface SuccessNotificationProps {
    props: NotificationOptions;
    method: "delete" | "post" | "put";
}

const successNotification = ({ props, method }: SuccessNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const customMessage = `api:${props.localeKey}.success.${method}.message`;
    const shouldOverideMessage = i18n.exists(customMessage);

    const message = shouldOverideMessage
        ? t(customMessage)
        : t(`api:common.success.${method}.message`, {
              item: props.itemName || "Item",
          });

    notificationService.success(message, { ...notificationProps });
};

export { errorNotification, successNotification };
