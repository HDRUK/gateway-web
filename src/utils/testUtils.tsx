import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
