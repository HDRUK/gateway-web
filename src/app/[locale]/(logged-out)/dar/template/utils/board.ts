import { BoardSections, Board, Task } from "../components";
import { BOARD_SECTIONS } from "../constants";
import { getTasksByStatus } from "./tasks";

export const initializeBoard = (tasks: Task[]) => {
    const boardSections: BoardSections = {};

    Object.keys(BOARD_SECTIONS).forEach(boardSectionKey => {
        boardSections[boardSectionKey] = getTasksByStatus(
            tasks,
            boardSectionKey as Board
        );
    });

    return boardSections;
};

export const findBoardSectionContainer = (
    boardSections: BoardSections,
    id: string
) => {
    if (id in boardSections) {
        return id;
    }

    const container = Object.keys(boardSections).find(key =>
        boardSections[key].find(item => item.id === id)
    );
    return container;
};
