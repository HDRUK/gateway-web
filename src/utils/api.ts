import { ReadonlyRequestCookies } from "next/dist/server/web/spec-extension/adapters/request-cookies";
import { Application } from "@/interfaces/Application";
import { AuthUser } from "@/interfaces/AuthUser";
import { CohortRequest } from "@/interfaces/CohortRequest";
import { ReducedCollection } from "@/interfaces/Collection";
import { DataCustodianNetwork } from "@/interfaces/DataCustodianNetwork";
import { DataUse } from "@/interfaces/DataUse";
import { Dataset } from "@/interfaces/Dataset";
import { Filter } from "@/interfaces/Filter";
import { FormHydrationSchema } from "@/interfaces/FormHydration";
import { NetworkSummary } from "@/interfaces/NetworkSummary";
import { GetOptions } from "@/interfaces/Response";
import { Team } from "@/interfaces/Team";
import { TeamSummary } from "@/interfaces/TeamSummary";
import { Tool } from "@/interfaces/Tool";
import { User } from "@/interfaces/User";
import apis from "@/config/apis";
import config from "@/config/config";
import { FILTERS_PER_PAGE } from "@/config/request";
import { getUserFromToken } from "@/utils/cookies";

async function get<T>(
    cookieStore: ReadonlyRequestCookies,
    url: string,
    options: GetOptions = {
        suppressError: false,
    }
): Promise<T> {
    const jwt = cookieStore.get(config.JWT_COOKIE);

    const res = await fetch(`${url}`, {
        headers: { Authorization: `Bearer ${jwt?.value}` },
    });

    if (!res.ok && !options.suppressError) {
        // This will activate the closest `error.js` Error Boundary
        throw new Error("Failed to fetch data");
    }

    const { data } = await res.json();

    return data;
}

async function getFilters(
    cookieStore: ReadonlyRequestCookies
): Promise<Filter[]> {
    return get<Filter[]>(
        cookieStore,
        `${apis.filtersV1UrlIP}?perPage=${FILTERS_PER_PAGE}`
    );
}

async function getEntityAction(
    cookieStore: ReadonlyRequestCookies
): Promise<string | undefined> {
    return cookieStore.get(config.ENTITY_ACTION_COOKIE.COOKIE_NAME)?.value;
}

function getUserFromCookie(cookieStore: ReadonlyRequestCookies): User | null {
    const jwt = cookieStore.get(config.JWT_COOKIE);

    return getUserFromToken(jwt?.value);
}

async function getUser(cookieStore: ReadonlyRequestCookies): Promise<AuthUser> {
    return get<AuthUser>(
        cookieStore,
        `${apis.usersV1UrlIP}/${getUserFromCookie(cookieStore)?.id}`
    );
}

async function getApplication(
    cookieStore: ReadonlyRequestCookies,
    applicationId: string
): Promise<Application> {
    return get<Application>(
        cookieStore,
        `${apis.applicationsV1UrlIP}/${applicationId}`
    );
}

async function getCohort(
    cookieStore: ReadonlyRequestCookies,
    cohortId: string
): Promise<CohortRequest> {
    return get<CohortRequest>(
        cookieStore,
        `${apis.cohortRequestsV1UrlIP}/${cohortId}`
    );
}

async function getTeam(
    cookieStore: ReadonlyRequestCookies,
    teamId: string
): Promise<Team> {
    const team = await get<Team>(cookieStore, `${apis.teamsV1UrlIP}/${teamId}`);

    return {
        ...team,
        users: team?.users.map(user => ({
            ...user,
            roles: user.roles.filter(
                // Remove global "hdruk" roles from team users
                role => !role.name.startsWith("hdruk")
            ),
        })),
    };
}

async function getTeamSummary(
    cookieStore: ReadonlyRequestCookies,
    teamId: string,
    options?: GetOptions
): Promise<TeamSummary> {
    return await get<TeamSummary>(
        cookieStore,
        `${apis.teamsV1UrlIP}/${teamId}/summary`,
        options
    );
}

async function getNetworkSummary(
    cookieStore: ReadonlyRequestCookies,
    networkId: string,
    options?: GetOptions
): Promise<NetworkSummary> {
    return await get<NetworkSummary>(
        cookieStore,
        `${apis.dataCustodianNetworkV1UrlIP}/${networkId}/summary`,
        options
    );
}

async function getDataCustodianNetworks(
    cookieStore: ReadonlyRequestCookies,
    networkId: string
): Promise<DataCustodianNetwork> {
    return await get<DataCustodianNetwork>(
        cookieStore,
        `${apis.dataCustodianNetworkV1UrlIP}/${networkId}`
    );
}

async function getDataset(
    cookieStore: ReadonlyRequestCookies,
    datasetId: string,
    schemaModel?: string,
    schemaVersion?: string,
    options?: GetOptions
): Promise<Dataset> {
    const baseUrl = `${apis.datasetsV1UrlIP}/${datasetId}`;
    const params = new URLSearchParams();

    if (schemaModel && schemaVersion) {
        params.append("schema_model", schemaModel);
        params.append("schema_version", schemaVersion);
    }
    const queryString = params.toString();

    return await get<Dataset>(
        cookieStore,
        queryString ? `${baseUrl}?${queryString}` : baseUrl,
        options
    );
}

async function getDataUse(
    cookieStore: ReadonlyRequestCookies,
    dataUseId: string,
    options?: GetOptions
): Promise<DataUse> {
    const dataUse = await get<DataUse[]>(
        cookieStore,
        `${apis.dataUseV1UrlIP}/${dataUseId}`,
        options
    );

    return dataUse?.[0];
}

async function getTool(
    cookieStore: ReadonlyRequestCookies,
    toolId: string,
    options?: GetOptions
): Promise<Tool> {
    const tool = await get<Tool>(
        cookieStore,
        `${apis.toolsV1UrlIP}/${toolId}`,
        options
    );

    return tool;
}

async function getReducedCollection(
    cookieStore: ReadonlyRequestCookies,
    collectionId: string,
    options?: GetOptions
): Promise<ReducedCollection> {
    const collection = await get<ReducedCollection>(
        cookieStore,
        `${apis.collectionsV1UrlIP}/${collectionId}?view_type=mini`,
        options
    );

    return collection;
}

async function getFormHydration(
    cookieStore: ReadonlyRequestCookies,
    schemaName: string,
    schemaVersion: string,
    dataTypes?: string[],
    teamId?: string
): Promise<FormHydrationSchema> {
    return get<FormHydrationSchema>(
        cookieStore,
        `${
            apis.formHydrationV1UrlIP
        }?name=${schemaName}&version=${schemaVersion}&dataTypes=${
            dataTypes || []
        }${teamId && `&team_id=${teamId}`}`
    );
}

export {
    getApplication,
    getCohort,
    getReducedCollection,
    getDataCustodianNetworks,
    getDataset,
    getDataUse,
    getEntityAction,
    getFilters,
    getFormHydration,
    getNetworkSummary,
    getTeam,
    getTeamSummary,
    getTool,
    getUser,
    getUserFromCookie,
};
