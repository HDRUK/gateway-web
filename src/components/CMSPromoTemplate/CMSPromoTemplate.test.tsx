import { render } from "@/utils/testUtils";
import CMSPromoTemplate from "./CMSPromoTemplate";

describe("CMSPromoTemplate", () => {
    const mockData = {
        id: "1",
        title: "Cohort Discovery Page",
        template: {
            promofields: {
                bannerTitle: "Cohort Discovery",
                ctaLink: {
                    title: "Button title",
                    url: "button/url",
                    target: "",
                },
                topRightPanel:
                    '<p><iframe loading="lazy" title="Cohort Discovery on the Health Data Research Innovation Gateway" width="500" height="281" src="https://www.youtube.com/embed/yvFrnbXlqRk?feature=oembed" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe></p>\n',
                topLeftPanel:
                    "<h2>About Cohort Discovery</h2>\n" +
                    '<p>Statistical disclosure control policies are in place for each data custodian, so low numbers are excluded from query results and results may also be rounded to eliminate any potential risk of identification.This functionality has been developed as part of the CO-CONNECT programme and further information can be found on their <a href="https://co-connect.ac.uk/">website</a>. The summary metadata of the current datasets that are available to query via Cohort Discovery can be found in this <a href="https://www.healthdatagateway.org/">Collection</a> on the Gateway.</p>\n',
                middlePanel:
                    "<h3>How you can request access to Cohort Discovery</h3>\n" +
                    '<p><img loading="lazy" decoding="async" class=" wp-image-130 aligncenter" src="https://storage.googleapis.com/hdrgatewaywordpresspoc/2023/12/e3b24faf-cohort-user-image-300x274.png" alt="Search and analyse medical groups" width="395" height="361" srcset="https://storage.googleapis.com/hdrgatewaywordpresspoc/2023/12/e3b24faf-cohort-user-image-300x274.png 300w, https://storage.googleapis.com/hdrgatewaywordpresspoc/2023/12/e3b24faf-cohort-user-image.png 601w" sizes="(max-width: 395px) 100vw, 395px" /></p>\n',
            },
        },
    };
    it("should render content", () => {
        const wrapper = render(<CMSPromoTemplate content={mockData} />);
        expect(wrapper.container).toMatchSnapshot();
    });
});
