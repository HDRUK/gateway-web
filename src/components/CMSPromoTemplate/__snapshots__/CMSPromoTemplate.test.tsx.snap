// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CMSPromoTemplate should render content 1`] = `
<div>
  <div
    class="wpStyles"
  >
    <div
      class="MuiBox-root mui-style-1ifzgud"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 mui-style-1j4mnqy-MuiTypography-root"
      >
        Cohort Discovery
      </h2>
    </div>
    <div
      class="MuiBox-root mui-style-1peaprr"
    >
      <div
        class="MuiBox-root mui-style-1t3222e"
      >
        <div
          class="MuiBox-root mui-style-e42i7k"
        >
          <div
            class="mui-style-65n8w-HTMLContent"
          >
            <h2>
              About Cohort Discovery
            </h2>
            

            <p>
              Statistical disclosure control policies are in place for each data custodian, so low numbers are excluded from query results and results may also be rounded to eliminate any potential risk of identification.This functionality has been developed as part of the CO-CONNECT programme and further information can be found on their 
              <a
                href="https://co-connect.ac.uk/"
              >
                website
              </a>
              . The summary metadata of the current datasets that are available to query via Cohort Discovery can be found in this 
              <a
                href="https://www.healthdatagateway.org/"
              >
                Collection
              </a>
               on the Gateway.
            </p>
            

          </div>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-135hpv8-MuiTypography-root-MuiLink-root"
            href="button/url"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium mui-style-1gtm4ny-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Button title
              <span
                class="MuiTouchRipple-root mui-style-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
        <div
          class="MuiBox-root mui-style-e42i7k"
        >
          <div
            class="mui-style-65n8w-HTMLContent"
          >
            <p>
              <iframe
                allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
                allowfullscreen=""
                frameborder="0"
                height="281"
                loading="lazy"
                src="https://www.youtube.com/embed/yvFrnbXlqRk?feature=oembed"
                title="Cohort Discovery on the Health Data Research Innovation Gateway"
                width="500"
              />
            </p>
            

          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root mui-style-1ivsh08"
    >
      <div
        class="MuiBox-root mui-style-1hvbzpf"
      >
        <div
          class="mui-style-65n8w-HTMLContent"
        >
          <h3>
            How you can request access to Cohort Discovery
          </h3>
          

          <p>
            <img
              alt="Search and analyse medical groups"
              class=" wp-image-130 aligncenter"
              decoding="async"
              height="361"
              loading="lazy"
              sizes="(max-width: 395px) 100vw, 395px"
              src="https://storage.googleapis.com/hdrgatewaywordpresspoc/2023/12/e3b24faf-cohort-user-image-300x274.png"
              srcset="https://storage.googleapis.com/hdrgatewaywordpresspoc/2023/12/e3b24faf-cohort-user-image-300x274.png 300w, https://storage.googleapis.com/hdrgatewaywordpresspoc/2023/12/e3b24faf-cohort-user-image.png 601w"
              width="395"
            />
          </p>
          

        </div>
      </div>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways mui-style-135hpv8-MuiTypography-root-MuiLink-root"
        href="button/url"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium mui-style-1gtm4ny-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Button title
          <span
            class="MuiTouchRipple-root mui-style-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>
</div>
`;
